python/modules/IcePy/Proxy.cpp

```diff
@@ -1561,7 +1562,7 @@ proxyIceFlushBatchRequestsAsync(ProxyObject* self, PyObject* /*args*/, PyObject*
         return nullptr;
     }
     callback->setFuture(future.get());
-    return future.release();
+    return IcePy::wrapFuture(*self->communicator, future.get());
```

Comment:
This is an unrelated fix. Code returning an future needs to call IcePy::wrapFuture to ensure the event loop adapter is used.